CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -133,174 +386,450 @@ public async Task GetStudentsResult()
+               .ReturnsAsync(new List<long>());
+
+            _dashboardRepositoryMock.Setup(x => x.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(new List<long>(), new List<long>()))
+                .ReturnsAsync(new List<AverageStudentMarkDto>());
+
+            InitializeForDashboardRepository();
+
+            var dashbordservice = new DashboardService(_unitOfWorkMock.Object, _currentUserServiceMock.Object);
+
+            //Act
+            var resultWithoutStudent = await dashbordservice.GetStudentsResultAsync(studentResultRequestWithOutStudent);
+
+            //Assert
+            resultWithoutStudent.Data.AverageStudentsMarks.Should().BeNullOrEmpty();
+        }
 
-            dashbordrepositoryMock.Setup(x => x.GetGroupsIdsByCourseIdAndPeriodAsync(
-                (long)studentResultRequestWithOutStudent.CourseId, 
+        [Fact]
+        public async Task GetStudentsResult_NotExistingCourseId_ShouldReturnEmptyStudentsAverageVisits()
+        {
+            //Arrange
+            var studentResultRequestWithOutStudent = new StudentsRequestDto<StudentResultType>()
+            {
+                CourseId = 5,
+                StartDate = new DateTime(2012, 1, 12),
+                FinishDate = new DateTime(2015, 4, 21),
+                IncludeAnalytics = new StudentResultType[]
+                {
+                    StudentResultType.AverageStudentMark,
+                    StudentResultType.AverageStudentVisits
+                }
+            };
+
+            _dashboardRepositoryMock.Setup(x => x.GetGroupsIdsByCourseIdAndPeriodAsync(
+                (long)studentResultRequestWithOutStudent.CourseId,
                 studentResultRequestWithOutStudent.StartDate,
                 studentResultRequestWithOutStudent.FinishDate))
-                .ReturnsAsync(new  List<long>());
+                .ReturnsAsync(new List<long>());
 
-            dashbordrepositoryMock.Setup(x => x.GetStudentsIdsByGroupIdsAsync(new List<long>() { 2 }))
-                .ReturnsAsync(new List<long>() { 6, 7, 8, 9, 10 });
+            _dashboardRepositoryMock.Setup(x => x.GetStudentsIdsByGroupIdsAsync(new List<long>()))
+               .ReturnsAsync(new List<long>());
+
+            _dashboardRepositoryMock.Setup(x => x.GetStudentsAverageVisitsByStudentIdsAndGroupsIdsAsync(new List<long>(), new List<long>()))
+                .ReturnsAsync(new List<AverageStudentVisitsDto>());
+
+            InitializeForDashboardRepository();
+
+            var dashbordservice = new DashboardService(_unitOfWorkMock.Object, _currentUserServiceMock.Object);
+
+            //Act
+            var resultWithoutStudent = await dashbordservice.GetStudentsResultAsync(studentResultRequestWithOutStudent);
+
+            //Assert
+            resultWithoutStudent.Data.AverageStudentVisits.Should().BeNullOrEmpty();
+        }
+
+        [Fact]
+        public async Task GetStudentClassbook_ValidDataPassed_ShouldReturnCurrentStudentMarks()
```

Comment:
Please test returned object for the success case in one method to simplify the configuration. Do this for all other tests.

Reply:
Solved like this https://github.com/ita-social-projects/WhatBackend/pull/409/commits/8f6a49149d50761f3377e73f420db31c754c8bce